api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/MessageProcessorScheduler.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.dhsc.htbhf.claimant.scheduler;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.message.MessageProcessor;
+import uk.gov.dhsc.htbhf.claimant.message.MessageType;
+import uk.gov.dhsc.htbhf.requestcontext.aop.NewRequestContextWithSessionId;
+
+/**
+ * Component responsible for scheduling the processing of messages.
+ * Each MessageType has its own cron schedule - see https://stackoverflow.com/a/26147143 for examples.
+ * Each messageType also has its own minimum and maximum lock period. For a definition of the ISO-8601 duration format, see
+ * https://www.digi.com/resources/documentation/digidocs/90001437-13/reference/r_iso_8601_duration_format.htm.
+ * Note that the {@link MessageProcessor} also specifies a limit on the number of messages to process
+ * in one go.
+ */
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class MessageProcessorScheduler {
+
+    private static final String EVERY_30_SECONDS = "*/30 * * * * *";
+    private static final String ONE_HOUR = "PT60M";
+    private static final String FIFTEEN_SECONDS = "PT15S";
+
+    private final MessageProcessor messageProcessor;
+
+    @Scheduled(cron = EVERY_30_SECONDS) 
+    @SchedulerLock(
+            name = "Process ADDITIONAL_PREGNANCY_PAYMENT messages",
+            lockAtLeastForString = FIFTEEN_SECONDS,
+            lockAtMostForString = ONE_HOUR)
+    @NewRequestContextWithSessionId(sessionId = "MessageProcessor:ADDITIONAL_PREGNANCY_PAYMENT")
+    public void processAdditionalPregnancyPaymentMessages() {
+        messageProcessor.processMessagesOfType(MessageType.ADDITIONAL_PREGNANCY_PAYMENT);
+    }
+
+    @Scheduled(cron = "*/30 * 8-18 * * *")  // every 30 seconds between 8am and 6pm
```

Comment:
I'm surprised the schedule isn't in config, is that going to be a subsequent PR?

Reply:
I wasn't sure if it needed to be in config. The question is, are we ever going to change this on a per-environment basis? If not, I'm not sure it needs to be in config.

Reply:
Yep, that's fine, we can change when we need to.